site/en/docs/extensions/mv3/nativeMessaging/index.md

```diff
@@ -0,0 +1,202 @@
+message:
+
+```js
+var port = chrome.runtime.connectNative('com.my_company.my_application');
+port.onMessage.addListener(function (msg) {
+  console.log('Received' + msg);
+});
+port.onDisconnect.addListener(function () {
+  console.log('Disconnected');
+});
+port.postMessage({text: 'Hello, my_application'});
+```
+
+[runtime.sendNativeMessage][send-native-message] can be used to send a message to native application without creating
+a port, e.g.:
+
+```js
+chrome.runtime.sendNativeMessage(
+  'com.my_company.my_application',
+  {text: 'Hello'},
+  function (response) {
+    console.log('Received ' + response);
+  }
+);
+```
+
+## Debugging native messaging {: #native-messaging-debugging }
+
+When the native messaging host fails to start, writes to `stderr` or when it violates the
+communication protocol, output is written to the error log of Chrome. On Linux and macOS, this log
+can easily be accessed by starting Chrome from the command line and watching its output in the
+terminal. On Windows, use `--enable-logging` as explained at [How to enable logging][enable-logging].
+
+Here are some errors and tips for solving the issues:
+
+- Failed to start native messaging host.
+  - Check whether you have sufficient permissions to execute the file.
+- Invalid native messaging host name specified.
+  - Check whether the name contains any invalid characters. Only lowercase alphanumeric characters,
+    underscores and dots are allowed. A name cannot start or end with a dot, and a dot cannot be
+    followed by another dot.
+- Native host has exited.
+  - The pipe to the native messaging host was broken before the message was read by Chrome. This is most
+    likely initiated from your native messaging host.
+- Specified native messaging host not found.
+  - Is the name spelled correctly in the extension and in the manifest file?
+  - Is the manifest put in the right directory and with the correct name? See [native messaging host
+    location][native-messaging-host-location] for the expected formats.
+  - Is the manifest file in the correct format? In particular, is the JSON syntax correct and do the
+    values match the definition of a [native messaging host manifest][native-messaging-host-manifest]?
+  - Does the file specified in `path` exist? On Windows, paths may be relative, but on macOS and Linux,
+    the paths must be absolute.
+- Native messaging host _host name_ is not registered. (Windows-only)
+  - The native messaging host was not found in the Windows registry. Double-check using `regedit`
+    whether the key was really created and matches the required format as documented at [native
+    messaging host location][native-messaging-host-location].
+- Access to the specified native messaging host is forbidden.
+  - Is the extension's origin listed in `allowed_origins`?
+- Error when communicating with the native messaging host.
+  - This is a very common error and indicates an incorrect implementation of the communication protocol
```

Comment:
If this is such a common error, why not list it first? Then you don't need to say it's a common error. You probably don't need to regardless.